File: resources/MTAW/server/popup.lua
--------------------------------------------------------------------------------
--<[ Модуль Popup ]>------------------------------------------------------------
--------------------------------------------------------------------------------
Popup = {
	init = function()
		
	end;
	
	-- Показать всплывающую подсказку
	-- > playerElement player - игрок, которому нужно показать подсказку
	-- > message string - сообщение подсказки
	-- > messageType string / nil - тип подсказки: info, warning, error, success (по умолчанию info)
	-- > icon string / nil - название значка из FontAwesome (по умолчанию зависит от типа сообщения)
	-- > delay number / nil - время (мс), которое будет видна подсказка (по умолчанию 5000)
	-- = void
	show = function( playerElement, message, messageType, icon, delay )
		triggerClientEvent( playerElement, "Popup.onServerShowPopup", resourceRoot, message, messageType, icon, delay )
	end;
};
addEventHandler( "onResourceStart", resourceRoot, Popup.init )

File: resources/MTAW/client/data/gui/js/popup.js
"use strict";
var Popup = {
	messageIcons : {
		info : 'info-circle',
		warning : 'exclamation-circle',
		error : 'exclamation-triangle',
		success : 'check-circle'
	},
	
	init : function() {
		
	},
	
	// Показать сообщение
	show : function( message, type, icon, delay ) {
		if ( typeof( type ) == 'undefined' ) {
			type = 'info';
		}
		if ( typeof( icon ) == 'undefined' ) {
			icon = Popup.messageIcons[ type ];
		}
		if ( typeof( delay ) == 'undefined' ) {
			delay = 5000;
		}
		
		var messageElement = $( '<div class="popup-message ' + type + '" />' );
		$( messageElement ).append( 
			$( '<table />' ).append(
				$( '<tr />' ).append(
					$( '<td class="message-icon" />' ).append( 
						$( '<i class="fa fa-' + icon + '" />' )
					),
					$( '<td class="message-content" />' ).html( message )
				)
			)
		);
		
		$( '#popup' ).append( messageElement );
		setTimeout( function() { Popup.hide( messageElement ); }, delay );
		setTimeout( function() { $( messageElement ).addClass( 'visible'); }, 20 );
		
		$( messageElement ).on( 'click', function() { Popup.hide( messageElement ); } );
		
		return messageElement;
	},
	
	hide : function( messageElement ) {
		$( messageElement ).removeClass( 'visible' );
		setTimeout( function() { $( messageElement ).remove(); }, 300 );
	}
};
$( document ).ready( function() {
	Popup.init();
} );